### ISSUE TITLE
Reliable mint event

### ISSUE THREAD

**Issue by metachris** [CityOfZion/neo-python#457]
Keeping this open for discussions about the `mint` event/

See also:

* https://github.com/CityOfZion/neo-python/pull/433#issuecomment-394789480
* https://github.com/CityOfZion/neo-python/pull/433#discussion_r192722975

**Comment by localhuman** [comment]
I've discussed this with others, and it is my opinion that the `transfer` event should not be hijacked to represent something that it isn't.  `transfer` is the transferral of value from one account to another, whereas `mint` is the creation of value.

Along with `mint`, there should be a corresponding `burn` event to be used when tokens are destroyed.

Neotracker.io is or is adding support for these events.

**Comment by brianlenz** [comment]
I agree @localhuman. Inferring minting behavior from `transfer` is problematic and has been implemented inconsistently in NEP-5 smart contracts to date. There's the question of how existing NEP-5 tokens should be handled on explorers, but that's really a separate discussion.

I think the problem here is that we're effectively establishing a new standard. These `mint` and `burn` events really feel like they should be in in NEP-5, but it doesn't look like they will be soon (https://github.com/neo-project/proposals/pull/44).

My thought is that `burn` could be optional (not all tokens need to support it), but every token has to be minted at some point, so it seems logical that `mint` would be part of the standard as opposed to the proposal of `transfer` from `null` being inferred as a mint.

Perhaps if NEP-5 isn't going to support a `mint` event, then neo-python shouldn't encourage use of a non-standard event? Of course, the same could be said for the `refund` and `approve` events, which similarly aren't in NEP-5...
